### PR TITLE
#7667 Added ability to view module loading warnings

### PR DESCRIPTION
**Pull request by ekozobrodov**
See [#7667](https://trac.genestack.com/ticket/7667)

Appropriate pull request in main repo: https://github.com/genestack/genestack/pull/3415

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,7 @@
+* 0.15.1
+    * Added support of 'warnings' to `status` command
```
Comment (vasiliy-bout) [inline ChangeLog]: What is the style in single quotes?

Comment (vasiliy-bout) [inline ChangeLog]: And by the way, maybe it's time to rename this file into one with proper extension? `ChangeLog.md`?

File: genestack_client/scripts/genestack_application_manager.py [hunk 2]
```diff
@@ -536,6 +533,15 @@ def upload_single_file(application, file_path, version, override,
     if not no_wait:
```
Comment (vasiliy-bout) [inline genestack_client/scripts/genestack_application_manager.py]: We added a message if we are waiting, but we should also add a message if we are not waiting.

If we are not waiting, we should remind users that if JAR was successfully loaded then there could be other errors and warnings that must be checked using `status` command manually by user.

File: genestack_client/scripts/genestack_application_manager.py [hunk 3]
```diff
@@ -536,6 +533,15 @@ def upload_single_file(application, file_path, version, override,
...
         for app_id in app_info.identifiers:
             wait_application_loading(application, app_id, version)
+        if app_info.identifiers:
+            # Get jar loading messages from one of its applications:
+            descriptor = application.invoke('getApplicationDescriptor', app_info.identifiers[0], version)
```
Comment (vasiliy-bout) [inline genestack_client/scripts/genestack_application_manager.py]: Reuse `get_application_descriptor`

Comment (vasiliy-bout) [inline genestack_client/scripts/genestack_application_manager.py]: We can reuse descriptor from `wait_application_loading`. Try to change method `wait_application_loading` into `do_when_application_loaded` that takes handler function as a parameter like in JS.

File: genestack_client/scripts/genestack_application_manager.py [hunk 4]
```diff
@@ -605,20 +611,37 @@ def wait_application_loading(application, app_id, version, seconds=1):
...
+            return False
     return True
 
 
+def format_loading_messages_by_lines(errors, warnings):
+    wrapper = TextWrapper(initial_indent='\t\t', subsequent_indent='\t\t', width=80)
+    lines = []
+    if warnings:
+        lines.append('\t%s' % 'Warnings:')
+        first = True
```
Comment (vasiliy-bout) [inline genestack_client/scripts/genestack_application_manager.py]: We can shrink this method to:
```python
def format_loading_messages_by_lines(errors, warnings):
    wrapper = TextWrapper(initial_indent='\t\t', subsequent_indent='\t\t', width=80)
    lines = []
    if warnings:
        lines.append('\t%s' % 'Warnings:')
        lines.append('\n'.join(['%s\n' % wrapper.fill(warning) for warning in warnings]))
    if errors:
        lines.append('\t%s' % 'Errors:')
        lines.append('\n'.join(['%s\n' % wrapper.fill(error) for error in errors]))
    return lines
```

File: ChangeLog [hunk 5]
```diff
@@ -1,3 +1,7 @@
+* 0.15.1
```
Comment (vasiliy-bout) [inline ChangeLog]: Fix ChangeLog version after merging with `master`

File: ChangeLog (hunk 1 above)
Comment (ekozobrodov) [inline ChangeLog]: I think it would be better to discuss it with @genestack-solomatin

File: genestack_client/version.py [hunk 6]
```diff
@@ -1,2 +1 @@
 __version__ = '0.16.0a1'
```
Comment (klas-genestack) [inline genestack_client/version.py]: Bump it to **a2**.

File: genestack_client/scripts/genestack_application_manager.py (hunk 3 above)
Comment (ekozobrodov) [inline genestack_client/scripts/genestack_application_manager.py]: I tried to implement waiting function with callback and this solution looks to complex and difficult for usage. So, i decided to simply return tuple with success flag and descriptor from `wait_application_loading` function.

**Pull request merged**